src/Client/Updater.php

```diff
@@ -45,25 +49,14 @@ public function __construct($repository_name, $mirrors)
   /**
    * @todo Update from python comment https://github.com/theupdateframework/tuf/blob/1cf085a360aaad739e1cc62fa19a2ece270bb693/tuf/client/updater.py#L999
    *
-   * @param bool $unsafely_update_root_if_necessary
    */
-    public function refresh($unsafely_update_root_if_necessary = true)
+    public function refresh()
     {
-    }
-
-  /**
-   * Validates a target.
-   *
-   * @param $target_repo_path
-   * @param $target_stream
-   *
-   * @return bool
-   *   Returns true if the target validates.
-   */
-    public function validateTarget($target_repo_path, $target_stream)
-    {
-        // @TODO source original "step 0" root data from client state, not repository
-        $root_data = json_decode($this->getRepoFile('root.json'), true);
+        // @TODO Inject DurableStorage
```

Comment:
>  How do we want to handle service injection?
As simply as possible :stuck_out_tongue: 

So far this is the only "injected" thing that we're planning on having the app provide to us. I think we should just ask for it in the `Updater` class's constructor and that's it.

In general I think hitching our wagon to a specific DI implementation in a library like this isn't a good idea because it couples it to a specific subset of the php ecosystem, like Symfony.

Reply:
I've been assuming that we could have the caller handle injection/services however they'd like by supply an implementation during construction. I'd prefer to not adopt a specific services configuration pattern within this library itself.